Graveyard UI not showing: MissingMethodException AnimatedCardList.Update

joc256 opened the pull request:
seems this method [is now async ](https://github.com/HearthSim/Hearthstone-Deck-Tracker/blame/master/Hearthstone%20Deck%20Tracker/Controls/AnimatedCardList.xaml.cs#L51)

I haven't tested this fix and I don't know C# but ChatGPT said it would work🤞 😅 
```
4:00:18 PM|Error|ActionList.Execute >> Error invoking action (Plugin: Graveyard):
System.MissingMethodException: Method not found: 'Void Hearthstone_Deck_Tracker.Controls.AnimatedCardList.Update(System.Collections.Generic.List`1<Hearthstone_Deck_Tracker.Hearthstone.Card>, Boolean)'.
   at HDT.Plugins.Graveyard.NormalView.Update(Card card)
   at HDT.Plugins.Graveyard.CardUpdatePoller.Poll(Card card)
   at Hearthstone_Deck_Tracker.API.ActionList`1.Execute(T arg) in D:\a\Hearthstone-Deck-Tracker\Hearthstone-Deck-Tracker\Hearthstone Deck Tracker\API\ActionList.cs:line 39

```

joc256 committed:
AnimatedCardList.Update is now async

joc256 committed:
AnimatedCardList.Update() now async

joc256 committed:
AnimatedCardList.Update() now async

joc256 committed:
AnimatedCardList.Update() now async

joc256 committed:
typo

joc256 committed:
typo

joc256 committed:
typo

p1r473 commented:
Can you build a new DLL? Im not sure how to compile

p1r473 commented:
Also how did you view the error log, Ive been wondering why the plugin hasnt been showing

batstyx commented:
> Also how did you view the error log, Ive been wondering why the plugin hasnt been showing

Logs are stored in `%appdata%\HearthstoneDeckTracker\Logs`

'%appdata%\HearthstoneDeckTracker\Logs\hdt_log.txt' is the "live" log

batstyx commented:
There's quite a lot of (incorrectly) moving parts here.
I'll take a look, but I can't guarantee a fast turnaround ¯\_(ツ)_/¯

batstyx committed:
HearthstoneDeckTracker.exe to v1.43.3.7018

batstyx commented:
If an HDT breaking change is being fixed, the repo copy of the HDT [exe](https://github.com/batstyx/Graveyard/blob/batstyx/lib/HearthstoneDeckTracker.exe) needs to updated using the [update-libraries](https://github.com/batstyx/Graveyard/blob/batstyx/scripts/update-libraries.ps1) PowerShell script and committed.
Not that that's helped here.

batstyx committed:
Update ReanimateView Catrina Muerte reference

batstyx committed:
ChancesTracker updated for AnimatedCardList changes
Reference AnimatedCards collection instead of Items collection from removed ItemsControl base

batstyx committed:
Correct AnimatedCardList UpdateAsync method calls to Update

batstyx commented:
I still don't think this is quite right, but it did at least build something testable.
If you click on the `Build+Upload / build-and-upload (pull_request)` link in the `All checks have passed` section, and then select `Summary` from the left-hand menu, you should see a test build to download in the Artifacts section.

batstyx committed:
Version to 1.18.1

batstyx commented:
Or alternatively, download the test build from the `Artifacts` dropdown of the [Checks](https://github.com/batstyx/Graveyard/pull/43/checks) tab at the top of this page.

p1r473 commented:
I'm getting this error and different ones also such as

```
11:54:01 AM|Error|CardUpdatePoller.Poll >> System.MissingMethodException: Method not found: 'Void Hearthstone_Deck_Tracker.Controls.AnimatedCardList.Update(System.Collections.Generic.List`1<Hearthstone_Deck_Tracker.Hearthstone.Card>, Boolean)'.
   at HDT.Plugins.Graveyard.NormalView.Update(Card card)
   at HDT.Plugins.Graveyard.CardUpdatePoller.Poll(Card card)

```

batstyx committed:
Remove .GetAwaiter().GetResult() hack
https://learn.microsoft.com/en-us/archive/msdn-magazine/2015/july/async-programming-brownfield-async-development#the-blocking-hack

batstyx commented:
Try [Graveyard-v1.18.1.2019](https://github.com/batstyx/Graveyard/suites/36721063697/artifacts/2885510303)

joc256 commented:
v1.18.1.2019 fixes the issue for me, thanks!

p1r473 commented:
Fixed, thanks
